### DISCUSSION TITLE
Template Widget__Input in Liquid -> How to access ModelState?

### DISCUSSION THREAD

**Discussion by bashuss** [OrchardCMS/OrchardCore#12016] [Q&A] [answered]
I am trying to override the Templates for the InputPart for a theme that works with liquid only.
What I need to achieve is a seperate handling for the inputs of type "checkbox" and "radio". This is especially about handling validation errors and cross browser styling.

To do it properly, I need to access the ModelState with liquid. How to do that?

The default razor template assigns `fieldEntry` with the data I would need:
`ViewData.ModelState.TryGetValue(fieldName, out var fieldEntry)`

I would need to access `fieldEntry.AttemptedValue` and `fieldEntry.Errors` to do it by just replacing the default template.

Anyone having different ideas or knowing how to access the ModelState?

**Comment by ns8482e** [comment, marked as answer]
I guess not every object is exposed to liquid - and IMHO liquid is not completely replaces `cshtml` and purpose of liquid is to provide safe design customization at run time. If you  need to more than just design then it's better to use `cshtml` - or register your own accessor for data that you want to expose in liquid.

Attempted value you can be accessed directly from request https://docs.orchardcore.net/en/latest/docs/reference/modules/Liquid/#request

**Reply by bashuss** [reply]
I would have hoped for something more useful in my case, but I guess you are right...
Thanks!

**Reply by bashuss** [reply]
In my case, the most practical way was to write a liquid filter in my own plugin. This filter exposes an object with the view data for a certain control if provided with a correct id, or if not provided with an id, it returns an array of available ids, so an iteration over the keys is possible (e.g. for an error message display).

Currently I exposed this in json, as it was the quickest way for me to ensure that no value can be edited or changed. Any direct exposing of the values would need more knowledge on the inner workings to keep it safe.

Any security concerns I missed?

**Reply by ns8482e** [reply]
I guess you are fine. Liquid is meant for design, It’s fine if you expose data that you read to design and only provide needed limited write back
